draft-ietf-httpbis-messaging-latest.xml

```diff
@@ -703,29 +703,22 @@ Host: www.example.org:8001
    the target URI's scheme is "https"; if not, the scheme is "http".
 </li>
 <li>
-   If the server's configuration (or outbound gateway) provides a fixed URI
-   <x:ref>authority</x:ref> component, that authority is used for the
-   target URI. If not, then if the request-target is in
-   <x:ref>authority-form</x:ref>, the target URI's authority
-   component is the same as the request-target.
-   If not, then if a <x:ref>Host</x:ref> header field is supplied with a
-   non-empty field value, the authority component is the same as the
-   Host field value. Otherwise, the authority component is assigned
-   the default name configured for the server and, if the connection's
-   incoming TCP port number differs from the default port for the target
-   URI's scheme, then a colon (":") and the incoming port number (in
-   decimal form) are appended to the authority component.
+   If the request-target is in <x:ref>authority-form</x:ref>,
+   the request-target is the target URI's authority component.
+   Otherwise, the target URI's authority component is the field value of the
+   <x:ref>Host</x:ref> header field, if that value is valid and non-empty, or
+   the authority component is empty if Host is absent, empty, or invalid.
 </li>
 <li>
    If the request-target is in <x:ref>authority-form</x:ref> or
    <x:ref>asterisk-form</x:ref>, the target URI's combined
-   <x:ref>path</x:ref> and <x:ref>query</x:ref> component is empty. Otherwise,
-   the combined <x:ref>path</x:ref> and <x:ref>query</x:ref> component is the
-   same as the request-target.
+   <x:ref>path</x:ref> and <x:ref>query</x:ref> component is empty.
+   Otherwise, the request-target is the target URI's combined
+   <x:ref>path</x:ref> and <x:ref>query</x:ref> component.
```

Comment:
I only just noticed this, but there is a difference in treatment between the items in this list.

The authority component uses "the authority component is X", whereas this uses "X is the combined path and query component".  Not sure if that is useful information or whether it just makes this harder to manage, but it does make reading this slightly harder than might be ideal.  (Though this change makes it a lot clearer.)

Reply:
Yeah, on third thought it is more consistent to describe it as variable assignment.